test/contracts/kill_switch/kill_switch.test.js

```diff
@@ -0,0 +1,963 @@
+const { skipCoverage } = require('../../helpers/coverage')
+const { assertRevert } = require('../../helpers/assertThrow')
+const { assertEvent, assertAmountOfEvents } = require('../../helpers/assertEvent')(web3)
+const { getNewProxyAddress, getEventArgument } = require('../../helpers/events')
+
+const ACL = artifacts.require('ACL')
+const Kernel = artifacts.require('Kernel')
+const DAOFactory = artifacts.require('DAOFactory')
+const KillSwitch = artifacts.require('KillSwitch')
+const IssuesRegistry = artifacts.require('IssuesRegistry')
+const KillSwitchedApp = artifacts.require('KillSwitchedAppMock')
+const EVMScriptRegistryFactory = artifacts.require('EVMScriptRegistryFactory')
+
+const FailingKillSwitchMock = artifacts.require('FailingKillSwitchMock')
+const KernelWithoutKillSwitchMock = artifacts.require('KernelWithoutKillSwitchMock')
+const KernelWithNonCompliantKillSwitchMock = artifacts.require('KernelWithNonCompliantKillSwitchMock')
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+contract('KillSwitch', ([_, root, owner, securityPartner, anyone]) => {
+  let dao, acl, app, registryFactory
+  let kernelBase, aclBase, appBase, killSwitchBase, issuesRegistryBase, daoFactory
+  let kernelWithoutKillSwitchBase, kernelWithNonCompliantKillSwitchBase, failingKillSwitchBase
+  let CORE_NAMESPACE, KERNEL_APP_ID, APP_MANAGER_ROLE, SET_SEVERITY_ROLE, SET_DEFAULT_ISSUES_REGISTRY_ROLE, SET_ISSUES_REGISTRY_ROLE, SET_ALLOWED_INSTANCES_ROLE, SET_DENIED_BASE_IMPLS_ROLE, SET_HIGHEST_ALLOWED_SEVERITY_ROLE
+
+  before('deploy base implementations', async () => {
+    // real
+    kernelBase = await Kernel.new(true) // petrify immediately
+    aclBase = await ACL.new()
+    registryFactory = await EVMScriptRegistryFactory.new()
+    killSwitchBase = await KillSwitch.new()
+    issuesRegistryBase = await IssuesRegistry.new()
+
+    // mocks
+    appBase = await KillSwitchedApp.new()
+    failingKillSwitchBase = await FailingKillSwitchMock.new()
+    kernelWithoutKillSwitchBase = await KernelWithoutKillSwitchMock.new()
+    kernelWithNonCompliantKillSwitchBase = await KernelWithNonCompliantKillSwitchMock.new()
+  })
+
+  before('load constants and roles', async () => {
+    CORE_NAMESPACE = await kernelBase.CORE_NAMESPACE()
+    KERNEL_APP_ID = await kernelBase.KERNEL_APP_ID()
+    APP_MANAGER_ROLE = await kernelBase.APP_MANAGER_ROLE()
+    SET_SEVERITY_ROLE = await issuesRegistryBase.SET_SEVERITY_ROLE()
+    SET_DEFAULT_ISSUES_REGISTRY_ROLE = await killSwitchBase.SET_DEFAULT_ISSUES_REGISTRY_ROLE()
+    SET_ISSUES_REGISTRY_ROLE = await killSwitchBase.SET_ISSUES_REGISTRY_ROLE()
+    SET_ALLOWED_INSTANCES_ROLE = await killSwitchBase.SET_ALLOWED_INSTANCES_ROLE()
+    SET_DENIED_BASE_IMPLS_ROLE = await killSwitchBase.SET_DENIED_BASE_IMPLS_ROLE()
+    SET_HIGHEST_ALLOWED_SEVERITY_ROLE = await killSwitchBase.SET_HIGHEST_ALLOWED_SEVERITY_ROLE()
+  })
+
+  context('when the kernel version does not support kill-switch logic', async () => {
```

Comment:
Doesn't this set of tests check a `KillSwitch`-compatible Kernel with no `KillSwitch` installed? I would say `KernelWithoutKillSwitchMock` is the one that best mimics the behaviour of old Kernels.

Reply:
Maybe the context message is misleading, but it is actually testing an "old kernel" version, that's why I mention that it "does not support kill-switch logic". Actually right below is a `beforeEach` statement deploying an "old" kernel using `KernelWithoutKillSwitchMock` (see [here](https://github.com/aragon/aragonOS/pull/518/files/ce7a29bf9d359abeab95c9a6b9075fe90ed8722d#diff-20fa58b7946252b898fd2dc455170cf6R56))

Reply:
🙏 I might've gotten confused with the context below, this test looks good!